Rewrite UCP backup/restore and improve uninstall and support guides

alexmavr opened the pull request:
Signed-off-by: Alex Mavrogiannis <alex.mavrogiannis@docker.com>

### Proposed changes

- Fixes https://github.com/docker/docker.github.io/issues/1751
- Fixes https://github.com/docker/docker.github.io/issues/1724
- Slightly touches https://github.com/docker/docker.github.io/issues/1750 but does not close it
- Renamed all "controller" occurences to "manager"
- Slightly rephrased a couple sentences here and there
- Added documentation for CLI support dumps on the "Get Support" page

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -14,27 +14,50 @@ The next step is creating a backup policy and disaster recovery plan.
-You need to run the backup command on a UCP manager node. Since UCP stores
-the same data on all manager nodes, you only need to create a backup of a
-single node.
+To create a UCP backup, you may use the `{{ page.docker_image }} backup` command
```
This is way more complete, but you sound like a boring lecturer in some sentences. Try writing like you speak.

What about

To create a backup of UCP, you can run the `{{ page.docker_image }} backup` command on a UCP manager node. <you don't need to tell me you're going to explain something below. If it's interesting I'm going to continue reading>
This command creates a tar archive...

alexmavr replied:
Understood. Please read my response to your comment in the next paragraph.

Oh wait.

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -14,27 +14,50 @@ The next step is creating a backup policy and disaster recovery plan.
+UCP](../architecture.md) to persist data and streams it to stdout.
+
+You only need to run the backup command on a single UCP manager node. Since UCP
+stores the same data on all manager nodes, you do not need to capture periodic
```
Maybe the other way around will be more obvious

"Since UCP stores the same data on all manager nodes, you only need to make periodic backups of a single node"

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -14,27 +14,50 @@ The next step is creating a backup policy and disaster recovery plan.
-containers and services are not affected by this.
+resources, such as services, containers and stacks are not affected by this
+operation and will continue operating as expected. Any long-lasting `exec`,
+`logs`, `events` or `attach` operations against the affected manager node will
```
Internally we use the word "against" a lot but when reading things for the first time, that word is a bit ambiguous. Maybe

"If you're running commands like `exec`, `logs`, `events`, or `attach` on a manager node, when performing a backup those commands will be disconnected."

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -14,27 +14,50 @@ The next step is creating a backup policy and disaster recovery plan.
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  {{ page.docker_image }} backup --interactive /tmp/backup.tar
+
+# Ensure the backup is a valid tar and list its contents
```
Can we include a bit more information on what to visually search for. I'll see a list of files, but I'll have no idea whether they contain all the data or not.

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -47,59 +70,63 @@ $ docker run --rm -i --name ucp \
+The restore command can be used to create a new UCP cluster from a backup file.
+After the restore operation is complete, the following data will be recovered
+from the backup file:
+* Users, Teams and Permissions.
```
We only capitalize product names, not concepts or features.

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -47,59 +70,63 @@ $ docker run --rm -i --name ucp \
+After the restore operation is complete, the following data will be recovered
+from the backup file:
+* Users, Teams and Permissions.
+* All UCP Configuration options available under `Admin Settings`, such as the
```
Configuration -> configuration
Subscription->subscription

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -47,59 +70,63 @@ $ docker run --rm -i --name ucp \
+DDC Subscription license, scheduling options, Content Trust and authentication
+backends.
+
+There restore operation can be performed in any of three environments:
```
There -> The
You say "three environments" but you only list two. Also you can make it a bit more straightforward like

"There are two ways to restore a UCP cluster"

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
````diff
@@ -47,59 +70,63 @@ $ docker run --rm -i --name ucp \
 ```
 
-## Restore your cluster
+## Disaster Recovery
````
Only capitalize the first word of a title (and product names)

joaofnfernandes commented on datacenter/ucp/2.1/guides/get-support.md:
```diff
@@ -17,8 +17,25 @@ Be sure to use your company email when filing tickets.
 2. On the top-right menu, **click your username**, and choose **Support Dump**.
+   An archive will be downloaded by your browser after a brief time interval.
+
+If the User Interface is not accessible, you may perform the following number of
```
No need to capitalize "user interface" too

alexmavr commented:
Added a commit which addresses comments and adds a few minor tweaks to the actual backup/restore commands that I discovered while actually going through the disaster recovery scenario

joaofnfernandes commented:
Everything looks good. Can you just address the comment on backups-and-disaster-recovery.md line 18?

alexmavr commented:
I did, not sure what you're referring to. I just amended the commit to rewrite that line word-for-word as per your recommendation, so I hope that checks out

vikstrous commented:
Does a ucp backup include a swarm backup? Also, by reading the swarm backup docs it's not clear to me what a swarm backup does.

alexmavr commented:
No, the UCP backup does not include a swarm backup. I think the swarm backup captures the set of services, node and CA keys and swarm-scoped configuration. The only actual things that is affected by UCP not performing a swarm backup is that services are not restored as part of the backup.

Given that it's possible to restore UCP on a different swarm cluster with a different set of services running, this might not be a feature worth pursuing. Additionally, instrumenting a swarm backup may prove to be challenging from UCP's perspective depending on whether the raft store should be unlocked for the backup or not.

joaofnfernandes commented:
That's an awesome question @vikstrous. Can we include something that states that the UCP backup command doesn't backup the cluster information like services, networks, ...
And maybe link to the swarm backup docs `/engine/swarm/admin_guide/#/back-up-the-swarm`

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -14,27 +14,52 @@ The next step is creating a backup policy and disaster recovery plan.
-To have minimal impact on your business, you should:
+Additionally, if UCP is not configured for high availability, you will be
+temporarily unable to:
+* Log in to the UCP Web UI
```
Our markdown renderer requires a space before list items 😓
![screen shot 2017-02-17 at 11 52 21](https://cloud.githubusercontent.com/assets/1525937/23080930/8ff62f92-f507-11e6-82a7-fe42fb4f472b.png)

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -14,27 +14,52 @@ The next step is creating a backup policy and disaster recovery plan.
+* Perform CLI operations using existing client bundles
 
-* Schedule the backup to take place outside business hours.
+To minimize the impact of the backup policy on your business, you should:
```
Add \n here too

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
````diff
@@ -14,27 +14,52 @@ The next step is creating a backup policy and disaster recovery plan.
+The example below shows how to create a backup of a UCP manager node and
+verify its contents:
+
+```bash
````
Our syntax highlighting is all crazy so change this to `none`.

![screen shot 2017-02-17 at 11 53 55](https://cloud.githubusercontent.com/assets/1525937/23080993/c9350a08-f507-11e6-8f31-dedd03690ac6.png)

joaofnfernandes commented on datacenter/ucp/2.1/guides/admin/backups-and-disaster-recovery.md:
```diff
@@ -47,71 +72,81 @@ $ docker run --rm -i --name ucp \
+DDC subscription license, scheduling options, Content Trust and authentication
+backends.
+
+There are two ways to restore a UCP cluster:
```
Add \n here too

alexmavr commented:
Addressed comments in new commit
